app/src/DataFileGenerator/FileGenerator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Phpsw\Website\DataFileGenerator;
+
+abstract class FileGenerator
+{
+    /**
+     * Use this function to obtain a slug for a given value (talk or event title, person's name etc).
+     * The slugs are file names, and they're also used to link the data together.
+     *
+     * This function replaces any special characters with a hyphen,
+     * replaces spaces with hyphens and lowercases the whole thing.
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    public function slugify(string $value): string
```

Comment:
I think we should favour composition over inheritance. E.g. have slugify as it's own method (maybe static method on a class) or injected in to the file generators. 
That way slugifier can be reused by things other than FileGenerators and also easy to add tests.

Reply:
We could do, I suppose slugify would be useful in URL / link construction too. @opdavies are you using anything for that currently?

I think a "utils class" with this function could be a good solution here, then can inject this class via DI.

Reply:
Yep. I like the utils class injected via DI solution.